src/algebra/monoid_algebra/basic.lean

```diff
@@ -542,14 +552,22 @@ lemma single_one_comm [comm_semiring k] [mul_one_class G] (r : k) (f : monoid_al
 by { ext, rw [single_one_mul_apply, mul_single_one_apply, mul_comm] }
 
 /-- `finsupp.single 1` as a `ring_hom` -/
-@[simps] def single_one_ring_hom [semiring k] [monoid G] : k →+* monoid_algebra k G :=
+@[simps] def single_one_ring_hom [semiring k] [mul_one_class G] : k →+* monoid_algebra k G :=
 { map_one' := rfl,
   map_mul' := λ x y, by rw [single_add_hom, single_mul_single, one_mul],
   ..finsupp.single_add_hom 1}
 
+/--  A multiplicative homomorphism `f : G →* H` between two monoids induces a ring
+homomorphism `monoid_ring_hom_map f : monoid_algebra k G →+* monoid_algebra k H`. -/
+def monoid_ring_hom_map (k : Type*) {H : Type*} [semiring k] [monoid G] [monoid H]
```

Comment:
I would guess you can make the semiring non-associative

Reply:
I am not sure that I know how to do this: I did not see ring homs with a non-associative assumption.  More specifically, I found results about additive maps (e.g. `lift_nc`), but multiplicativity seems to not be bundled, as I also found `lift_nc_mul` and `lift_nc_one`.

I probably prefer to leave this to someone else, since for the application to Laurent polynomials, I do not actually need it.

Reply:
I tried, but `monoid_algebra` wants a semiring.  Possibly I may have to split `k` into a `comm_semiring` and a `non_assoc_ring R` on top... maybe.

I would leave it as is.